File: backend-nestjs/src/modules/auth/dto/password-reset-response.dto.ts
export class PasswordResetResponseDto {
  success!: boolean;
  message!: string;
  expiresInMinutes?: number;

  static success(message: string, expiresInMinutes?: number): PasswordResetResponseDto {
    return {
      success: true,
      message,
      expiresInMinutes,
    };
  }

  static error(message: string): PasswordResetResponseDto {
    return {
      success: false,
      message,
    };
  }

  static builder() {
    return new PasswordResetResponseBuilder();
  }
}

class PasswordResetResponseBuilder {
  private dto: PasswordResetResponseDto = { success: false, message: '' };

  success(value: boolean): this {
    this.dto.success = value;
    return this;
  }

  message(value: string): this {
    this.dto.message = value;
    return this;
  }

  expiresInMinutes(value: number): this {
    this.dto.expiresInMinutes = value;
    return this;
  }

  build(): PasswordResetResponseDto {
    return this.dto;
  }
}

File: backend/src/main/java/com/northwollo/tourism/dto/response/PasswordResetResponseDto.java
package com.northwollo.tourism.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PasswordResetResponseDto {

    private String message;
    private boolean success;
    private Integer expiresInMinutes; // OTP expiry time

    public static PasswordResetResponseDto success(String message) {
        return PasswordResetResponseDto.builder()
                .message(message)
                .success(true)
                .build();
    }

    public static PasswordResetResponseDto error(String message) {
        return PasswordResetResponseDto.builder()
                .message(message)
                .success(false)
                .build();
    }
}